src/test/unit_tests/test_bucketspace_versioning.js

```diff
@@ -2664,6 +2664,24 @@ mocha.describe('bucketspace namespace_fs - versioning', function() {
                 assert.fail(`Failed with an error: ${err.Code}`);
             }
         });
+
+        mocha.it('head object, with version enabled, version id specified delete marker - should throw error with code 405', async function() {
+            try {
+                await s3_client.headObject({Bucket: bucket_name, Key: en_version_key, VersionId: versionID_1});
+                assert.fail('Should fail');
+            } catch (err) {
+                assert.strictEqual(err.$metadata.httpStatusCode, 405);
```

Comment:
add here a comment about what we saw on AWS CLI HEAD and GET different results
and on the get check also we got methodNotAllowed

Reply:
I added comments in both cases and attached the AWS CLI output, thanks.